app/controllers/admin/site_steps_controller.rb

```diff
@@ -157,6 +157,11 @@ def update
         else
           @site.form_step = 'template'
 
+          if @site.site_template.nil?
```

Comment:
this should ideally be happening as part of validations within the model, would that work?